### PR TITLE
Remove a bunch of old samples.

### PR DESCRIPTION
**Pull request by fundthmcalculus**
As part of cleaning up the samples, let's remove some we don't use that are duplicated in the `sdk-examples` repository.

### CODE REVIEW

File: samples/dotnet/Program.cs [hunk 1]
```diff
@@ -1,111 +0,0 @@
-﻿using System;
```
Comment (janpieterz) [inline samples/dotnet/Program.cs:1]: This feels like a pretty useful sample actually for maybe all languages to have? Something simple like this showing a flow from a-z? Why would this be deleted?

Comment (janpieterz) [reply samples/dotnet/Program.cs:1]: Or are these replicated in `sdk-examples`?

Comment (fundthmcalculus) [reply samples/dotnet/Program.cs:1]: They're replicated in `sdk examples`

Comment (fundthmcalculus) [reply samples/dotnet/Program.cs:1]: Now that we have cross repository action calls, it's possible to kick off the SDK examples pipeline whenever we make an SDK release

File: .github/workflows/build-golang.yml [hunk 2]
```diff
@@ -21,11 +21,11 @@ on:
 jobs:
   build-and-test-golang:
     name: Build, Test
-    runs-on: ${{ matrix.os-artifact[0] }}
+    runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false
       matrix:
-        os-artifact: [ [ubuntu-latest, linux], [windows-latest, windows], [macos-latest, macos] ]
+        os: [ ubuntu-latest, windows-latest, macos-latest ]
```
Comment (janpieterz) [inline .github/workflows/build-golang.yml:28]: Why no more Linux? Feels like a different change than removing samples.

**Review by janpieterz — APPROVED** [review]

File: .github/workflows/build-golang.yml (hunk 2 above)
Comment (fundthmcalculus) [reply .github/workflows/build-golang.yml:28]: `ubuntu-latest` is linux. I am removing the unnecessary artifact download, not the os...

**Pull request merged**